### PR TITLE
S3: clean-up multiple upload branch

### PR DESCRIPTION
**Pull request by JadenSimon**
Mostly focused on general code (and telemetry) clean-up. Not personally a big fan of having a progress notification for every single upload. ~Thinking we just coalesce them into the 'main' progress? I believe David wanted to be able to cancel individual uploads but most things get uploaded so fast you'd never be able to react.~ Also didn't spend too much time on refactoring since I kind of need some common UI elements to work with.

Updates to the original PR:
* Removed the extra progress notification. It's now just a single one.
* Show the size of the current file being uploaded, as well as the progress for larger files
* Changed `updateObjects` to `updateObject` telemetry with failed/success counts. Retrying the upload counts as a separate metric.

Future work:
* Don't use `withProgress` here. Outputting every file upload to the output channel can be noisy.

Other than that the feature works.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.nls.json [hunk 1]
```diff
@@ -437,11 +437,16 @@
...
     "AWS.s3.promptTime.placeHolder": "Defaults to 15 minutes",
     "AWS.s3.promptTime.prompt": "Enter a valid time (in minutes) for URL: ${0}",
+    "AWS.s3.uploadFile.progressTitle.batch": "Uploading {0} files to {1}",
     "AWS.s3.uploadFile.progressTitle": "Uploading {0}...",
+    "AWS.s3.uploadFile.progressReport": "{0}/{1} file(s) uploaded to {3}",
     "AWS.s3.uploadFile.openButton": "Upload",
     "AWS.s3.uploadFile.startUpload": "Uploading file from {0} to {1}",
-    "AWS.s3.uploadFile.success": "Successfully uploaded file {0} to {1}",
-    "AWS.s3.uploadFile.error.general": "Failed to upload file {0}",
+    "AWS.s3.uploadFile.success": "Successfully uploaded {0}/{1} file(s)",
```
Comment (justinmk3) [inline package.nls.json:445]: "Successfully" is noise here.
```suggestion
    "AWS.s3.uploadFile.success": "Uploaded {0}/{1} file(s)",
```

File: package.json [hunk 2]
```diff
@@ -2373,7 +2373,7 @@
         "serve": "webpack serve --config-name vue-hmr --mode development"
     },
     "devDependencies": {
-        "@aws-toolkits/telemetry": "1.0.21",
+        "@aws-toolkits/telemetry": "1.0.27",
```
Comment (justinmk3) [inline package.json:2376]: After https://github.com/aws/aws-toolkit-common/pull/248 (1.0.28) do we need another change in this PR?

**Review by justinmk3 — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (JadenSimon) [reply package.json:2376]: Yeah it's 10.0.27

**Comment by JadenSimon** [comment]
I have one tweak for showing the current/total bytes for the ongoing upload but then I'll merge w/ changelog.

**Comment by JadenSimon** [comment]
There's a few things (a little bit UI, a little bit code) that can be cleaned up here but I'll save that for another day.

**Pull request merged**